service/history/handler.go

```diff
@@ -2175,7 +2174,11 @@ func (h *handlerImpl) error(
 		// We will delete the workflow or mark the workflow as corrupted.
 		// Placing a dummy call to the function to check the coherency of the design.
 		// The function returns nil error so removing error handling for now.
-		h.GetTaskValidator().WorkflowCheckforValidation(workflowID, domainID, "")
+		domainName, err := h.GetDomainCache().GetDomainName(domainID)
+		if err != nil {
+			domainName = ""
```

Comment:
It should be rare to have a domain retrieval from cache fail. Ignoring the error and moving on with empty domain name is going to introduce yet another caveat to the system. 
I'd return the `err` here instead of moving on with undefined domain name. In general, "fail fast" approach helps eliminate bugs by eliminating bunch of undefined behavior